components/installer/pkg/steps/remove-kyma-components.go

```diff
@@ -22,9 +25,15 @@ func (steps InstallationSteps) RemoveKymaComponents() {
 	log.Println("All Kyma resources have been successfully removed")
 }
 
-func (steps InstallationSteps) uninstallReleases() error {
+func (steps InstallationSteps) uninstallReleases(installationData *config.InstallationData) error {
 	log.Println("Uninstalling releases...")
-	releasesToBeDeleted := []string{"ec-default", "hmc-default", "dex", "core", "prometheus-operator", "istio", "cluster-essentials"}
+
+	releasesRes, err := steps.helmClient.ListReleases()
+	if err != nil {
```

Comment:
You can simply convert the response from helm list into map[string]bool where key is the release name (value is not relevant here, can be bool). Then iterate over the installationData.Components and check if the release from component is installer (exists in map). If no, then log this fact, otherwise perform DeleteRelease action.

Reply:
+1. To maximize performance, a string-struct map (map[string]struct{}) should be your first choice.